backup.rb

```diff
@@ -18,13 +18,25 @@
 time = Time.now.strftime("%Y-%m-%d")
 filename = "backup.#{Time.now.to_i}.#{time}.sql.dump"
 
-`pg_dump -Fc --username=#{pg_user} --no-password --host #{pg_host} --port #{pg_port} #{pg_database} > #{filename}`
+if pg_user && pg_host
+  `pg_dump -Fc --username=#{pg_user} --no-password --host #{pg_host} --port #{pg_port} #{pg_database} > #{filename}`
+elsif pg_host
+  `pg_dump -Fc --host #{pg_host} --port #{pg_port} #{pg_database} > #{filename}`
+else
+  `pg_dump -Fc #{pg_database} > #{filename}`
+end
 
 # verify file exists and file size is > 0 bytes
 unless File.exists?(filename) && File.new(filename).size > 0
   raise "Database was not backed up"
 end
 
+if ENV["ENCRYPT_TO"]
```

Comment:
probably. In the case I wrote this for we are doing all of this within an encrypted partition, so it didn't really matter that there was an inflight unencrypted file.  Seems like a reasonable change if gpg can do it.